test/fuzz/transformation_replace_id_with_synonym_test.cpp

```diff
@@ -1888,7 +1888,7 @@ TEST(TransformationReplaceIdWithSynonymTest,
 //  test so that it covers more atomic operations, and enable the test once the
 //  issue is fixed.
 TEST(TransformationReplaceIdWithSynonymTest,
```

Comment:
Because this PR addresses the above TODO comment, the TODO comment should be removed as part of the PR.

Reply:
Done.

Reply:
Actually, I think now that 4349 is merged this test will no longer pass: 4349 will stop the synonym replacements being possible due to the rules on constants for the Shader capability.

Please confirm this by updating master and rebasing, and then seeing whether this fails. If it does fail, please put back the comment about the test being disabled and make the test disabled again.

If it passes then let's discuss (because I don't think it should).

Reply:
@afd Test `SignOfAtomicScopeAndMemorySemanticsDoesNotMatter` fails.